File: cpp/space-age/space_age.cpp
#include "space_age.h"

        space_age::space_age::space_age(long sec_in) {
            second = sec_in;
        }

         double space_age::space_age::seconds() const{
            return second;
        }

        double space_age::space_age::on_earth() const {
            return  second/31557600.0 ;
        }

        double space_age::space_age::on_mercury() const {
            return second/7600543.81992;
        }

        double space_age::space_age::on_venus() const {
            return second/19414149.052;
        }

        double space_age::space_age::on_mars() const {
            return second/59354032.69;
        }

        double space_age::space_age::on_jupiter() const {
            return second/374355659.124;
        }

        double space_age::space_age::on_saturn() const {
            return second/929292362.8848;
        }

        double space_age::space_age::on_uranus() const {
            return second/2651370019.3296;
        }

        double space_age::space_age::on_neptune() const {
            return second/5200418560.032;
        }

File: csharp/space-age/SpaceAge.cs
using System;

public class SpaceAge
{
    private int second;
    public SpaceAge(int seconds)
    {
        second = seconds;
    }

    public double OnEarth()
    {
        return second / 31557600.0;
    }

    public double OnMercury()
    {
        return second / 7600543.81992;
    }

    public double OnVenus()
    {
        return second / 19414149.052; ;
    }

    public double OnMars()
    {
        return second / 59354032.69;
    }

    public double OnJupiter()
    {
        return second / 374355659.124;
    }

    public double OnSaturn()
    {
        return second / 929292362.8848;
    }

    public double OnUranus()
    {
        return second / 2651370019.3296;
    }

    public double OnNeptune()
    {
        return second / 5200418560.032;
    }
}
